src/core/featuremodel.cpp

```diff
@@ -138,7 +140,24 @@ void FeatureModel::setCurrentLayer( QgsVectorLayer *layer )
       ( *sRememberings )[mLayer].rememberedFeature = mFeature;
       ( *sRememberings )[mLayer].rememberedAttributes.fill( false, layer->fields().size() );
     }
+
+    mGeometryLockedByDefault = mLayer->customProperty( QStringLiteral( "QFieldSync/is_geometry_locked_expression" ), false ).toBool();
+    const bool geometryLockedExpressionActive = mLayer->customProperty( QStringLiteral( "QFieldSync/is_geometry_locked_expression_active" ), false ).toBool();
+    if ( geometryLockedExpressionActive )
```

Comment:
```suggestion
    const QString geometryLockedExpressionActive = mLayer->customProperty( QStringLiteral( "QFieldSync/is_geometry_locked_expression_active" ) ).toString();
    if ( !geometryLockedExpressionActive.trimmed().isEmpty() )
```

I think this is a bit more precise what we want?

Reply:
@suricactus , no, in this case the active is a boolean, vs. the expression itself which is a string.